src/app/Ranking.js

```diff
@@ -0,0 +1,57 @@
+export class Ranking {
+    constructor(mode) {
+        const availableModes = ['people', 'starships', 'vehicles'];
+        if (!availableModes.includes(mode)) throw `"${mode}" mode is not valid!`;
+        this.mode = mode;
+    }
+
+    saveScore(user, score, maxScore) {
+        if (this._isLessThanThreeResults()) {
+            const topResults = JSON.parse(localStorage.getItem(this.mode));
+            topResults.push({
+                user,
+                score,
+                maxScore
+            });
+            localStorage.setItem(this.mode, JSON.stringify(topResults));
+        } else if (this._isHigherThanCurrentTop(score, maxScore)) {
+            const topResults = this._getTwoHighestResults();
+            topResults.push({
+                user,
+                score,
+                maxScore
+            });
+            localStorage.setItem(this.mode, JSON.stringify(topResults));
+        }
+    }
+    _isLessThanThreeResults() {
+        let results = localStorage.getItem(this.mode);
+        if (!results) {
+            localStorage.setItem(this.mode, JSON.stringify([]));
+            results = localStorage.getItem(this.mode);
+        };
+        const numOfResults = JSON.parse(results).length;
+        return numOfResults < 3;
+    }
+
+    _isHigherThanCurrentTop(score, maxScore) {
+        const topResults = JSON.parse(localStorage.getItem(this.mode));
+        const totalScores = topResults.map(result => result['score'] / result['maxScore']);
+        return score / maxScore >= Math.min(...totalScores);
+    }
+
+    _getTwoHighestResults() {
+        const topResults = JSON.parse(localStorage.getItem(this.mode));
+        const totalScores = topResults.map(result => result['score'] / result['maxScore']);
+        return topResults.filter((result, i) => i !== totalScores.indexOf(Math.min(...totalScores)));
+    }
+
+    getScores() {
```

Comment:
Teraz wszystko jest OK :) 

Chociaż ostatni improvement:
Tutaj jeszcze dość ważna zasada. O ile nam nie zaszkodzi w tym przypadku to czasem może. Warto, żeby metoda była jednego typu wg. konwencji CQS (Command-Query Separation), czyli albo odpowiada na pytanie albo modyfikuje stan. 
Przy getScores() spodziewamy sie po prostu, ze zwroci liste wynikow, a jednak cos tutaj modyfikujemy (zapisujemy pusta tablice). Mysle, ze mozna to zrobic tez prosciej:
```
getScores() {
     return localStorage.getItem(this.mode) ?? [];
}
```

Reply:
O matko, faktycznie. Mówiłam, że muszę jeszcze dzisiaj spojrzeć na to świeżym okiem :)